File: user-service/src/main/java/com/gallery/userservice/service/AuthService.java
package com.gallery.userservice.service;

import com.gallery.userservice.domain.Auth;
import com.gallery.userservice.domain.IAuthRepository;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.stereotype.Service;

@Service
public class AuthService {

    private final IAuthRepository authRepository;
    private final BCryptPasswordEncoder encoder;

    public AuthService(IAuthRepository authRepository) {
        this.authRepository = authRepository;
        this.encoder = new BCryptPasswordEncoder();
    }

    public boolean signUp(Auth auth) {
        if (auth.getEmail() == null || auth.getPasswordHash() == null)
            return false;


        String hashedPassword = encoder.encode(auth.getPasswordHash());
        Auth hashedAuth = new Auth(0, auth.getUserId(), auth.getEmail(), hashedPassword);

        return authRepository.insert(hashedAuth);
    }

    public boolean login(String email, String password) {
        Auth auth = authRepository.findByEmail(email);
        if (auth == null) return false;

        return encoder.matches(password, auth.getPasswordHash());
    }
}

File: app/UserService/Services/AuthService.cs
using Org.BouncyCastle.Crypto.Generators;
using UserService.Domain;
using UserService.Domain.Contracts;

namespace UserService.Services
{
    public class AuthService
    {
        private IAuthDAO _authDAO;

        public AuthService(IAuthDAO authDAO)
        {
            this._authDAO = authDAO;
        }

        public bool SignUp(Auth auth)
        {
            if (auth == null)
                return false;

            // Hash parola
            string hashedPassword = BCrypt.Net.BCrypt.HashPassword(auth.PasswordHash);
            auth.PasswordHash = hashedPassword;

            return _authDAO.SignUp(auth);
        }

        public Auth? LogIn(string email, string passwordHash)
        {
            if (string.IsNullOrEmpty(email) || string.IsNullOrEmpty(passwordHash))
                return null;
            return _authDAO.LogIn(email, passwordHash);
        }

        public bool DeleteAuthByUserId(int userId)
        {
            if (userId <= 0)
                return false;
            return _authDAO.DeleteAuthByUserId(userId);
        }


        public Auth? GetAuthByEmail(string email)
        {
            if (string.IsNullOrEmpty(email)) return null;
            return _authDAO.GetAuthByEmail(email);
        }

    }
}
